programs/vote/src/vote_instruction.rs

```diff
@@ -199,8 +199,8 @@ pub fn vote(vote_pubkey: &Pubkey, authorized_voter_pubkey: &Pubkey, vote: Vote)
         AccountMeta::new(*vote_pubkey, false),
         AccountMeta::new_readonly(sysvar::slot_hashes::id(), false),
         AccountMeta::new_readonly(sysvar::clock::id(), false),
-    ]
-    .with_signer(authorized_voter_pubkey);
+        AccountMeta::new_readonly(*authorized_voter_pubkey, true),
```

Comment:
This exposed a bug in the message compiler. If `vote_pubkey` and `authorized_voter_pubkey` are the same, then the key should be promoted to writable.

Reply:
Fixed in 9753d242174f1e6396c8f681c72b344d1811c46e